File: Homework1Java/src/task5/Folder.java
package task5;

import java.util.ArrayList;
import java.util.Collections;

public class Folder extends AbstractFileSystemNode{
    private ArrayList<FileSystemNode> list = new ArrayList<>();

    public Folder(String name, FileSystemNode ... fileSystemNodes){
        this.name = name;
        Collections.addAll(list, fileSystemNodes);
    }

    public void add(FileSystemNode fileSystemNode){
        list.add(fileSystemNode);
    }

    public ArrayList<FileSystemNode> getList() {
        return list;
    }

    @Override
    public String toString() {
        return name + ": " + list;
    }
}

File: Homework2Kotlin/src/task3/Folder.kt
package task3

class Folder(name: String?, vararg fileSystemNodes: FileSystemNode?) : AbstractFileSystemNode() {
    var list = mutableListOf(*fileSystemNodes)

    init {
        this.name = name
    }

    fun add(fileSystemNode: FileSystemNode) {
        list.add(fileSystemNode)
    }

    override fun toString(): String {
        return "$name: $list"
    }
}